Objects/unicodeobject.c

```diff
@@ -6086,17 +6082,13 @@ _PyUnicode_DecodeUnicodeEscape(const char *s,
 
       error:
         endinpos = s-starts;
-        writer.min_length = end - s + writer.pos;
         if (unicode_decode_call_errorhandler_writer(
                 errors, &errorHandler,
                 "unicodeescape", message,
                 &starts, &end, &startinpos, &endinpos, &exc, &s,
                 &writer)) {
             goto onError;
         }
-        if (_PyUnicodeWriter_Prepare(&writer, writer.min_length, 127) < 0) {
-            goto onError;
```

Comment:
I cannot understand. :-( `writer.min_length` means the least needed space here and it means the least space `_PyUnicodeWriter` will allocate for you. And `writer.size` is the actually allocated size. So shouldn't the right assertion here is just `assert(writer.min_length <= writer.size)`. If you minus `writer.pos`, it means the left space, then should use `assert(end - s <= writer.size - writer.pos)`.